File: tool/table/client/buff.ts
/*
Author: 
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
*/

module config{

let buff_map = null;
export function buff_map_init(config_obj:Object):void{
	buff_map = config_obj["buff_map"];
}


export class Buff extends Object{
	private m_config:any;
	public static m_static_map:Object = new Object();
	constructor(key){
		super();
		this.m_config = buff_map[key];
		for(let i in this.m_config){
			this[i] = this.m_config[i];
		}
		return;
	}
	public static get_Buff(key){
		if(Buff.m_static_map.hasOwnProperty(key) == false){
			Buff.m_static_map[key] = Buff.create_Buff(key);
		}
		return Buff.m_static_map[key];	
	}
	public static create_Buff(key){
		if(buff_map.hasOwnProperty(key)){
			return new Buff(key);
		}
		return null;	
	}
	public static get_cfg_object():Object
	{
		return buff_map;
	}
}
}

File: app/config/buff.py
# -*- coding: utf-8 -*-

'''
Author: Hannibal
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
'''


buff_map = {};
buff_map[1001] = {"id":1001,"btype":0,"effect":"actor['hpmax']*=1.2","group":1,};
buff_map[1002] = {"id":1002,"btype":0,"effect":"actor['spd']+=20","group":1,};
buff_map[1003] = {"id":1003,"btype":0,"effect":"actor['atk']+=20","group":1,};
buff_map[1004] = {"id":1004,"btype":0,"effect":"actor['def']+=20","group":1,};


class Buff:
	def __init__(self, key):
		config = buff_map.get(key);
		for k, v in config.items():
			setattr(self, k, v);
		return

def create_Buff(key):
		config = buff_map.get(key);
		if not config:
			return
		return Buff(key)

